internal/services/azurestackhci/stack_hci_deployment_setting_resource.go

```diff
@@ -0,0 +1,1365 @@
+														"Compute",
+														"Storage",
+														"Management",
+													}, false),
+												},
+											},
+
+											"adapter_property_override": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"jumbo_packet": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validation.StringIsNotEmpty,
+														},
+
+														"network_direct": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validation.StringIsNotEmpty,
+														},
+
+														"network_direct_technology": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validation.StringIsNotEmpty,
+														},
+													},
+												},
+											},
+
+											"adapter_property_override_enabled": {
+												Type:     pluginsdk.TypeBool,
+												Optional: true,
+												ForceNew: true,
+												Default:  false,
+											},
+
+											"qos_policy_override": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"bandwidth_percentage_smb": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validation.StringIsNotEmpty,
+														},
+
+														"priority_value8021_action_cluster": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validation.StringIsNotEmpty,
+														},
+
+														"priority_value8021_action_smb": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validation.StringIsNotEmpty,
+														},
+													},
+												},
+											},
+
+											"qos_policy_override_enabled": {
+												Type:     pluginsdk.TypeBool,
+												Optional: true,
+												ForceNew: true,
+												Default:  false,
+											},
+
+											"virtual_switch_configuration_override": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"enable_iov": {
+															Type:         pluginsdk.TypeString,
```

Comment:
this is an "enable" so true or false but is a string? shoiuld it not be a boolean ?

Reply:
It's plain string format and related to specific OEM setting. And by test, empty string should be sent if user does not specify the property. https://github.com/Azure/azure-rest-api-specs/blob/c41b76298becce0a8cfac4e682773d5d0c78dcb2/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2024-01-01/deploymentSettings.json#L784-L786